### ISSUE TITLE
[FEATURE] Weapon Mods Should Affect Weapons

### ISSUE THREAD

**Issue by Grygon** [Eranziel/foundryvtt-lancer#146]
**Is your feature request related to a problem? Please describe.**
Currently, when you apply a weapon mod it doesn't actually combine into an "altered weapon" or anything

**Describe the solution you'd like**
Weapon mod should automatically apply bonuses, tags, actions, etc

**Describe alternatives you've considered**
Manually adding it to the weapon--works, but sucks

**Additional context**
Would be cool

**Comment by msprijatelj** [comment]
So this looks like Lancer v1.4.1 displays everything correctly at the very least:

- Tags add correctly (including "Knockback" adding up numerically, assuming the tags use numbers and not strings). 
- Bonuses don't seem to automatically add (namely the Range Bonus for Stabilizer Mod) but are displayed among the details of the mod.
- Actions for a mod (See Shock Wreath) display correctly, but the action button throws an error due to it not being a "System" or a "Feature" (meaning that we probably need to add "Weapon Mod" handling for it).

Basically, everything is displayed, though some extra calculations (bonuses) and bug fixes (action buttons) could be performed.

![image](https://user-images.githubusercontent.com/44978587/213793648-21e5a506-7df2-402a-ac5d-596cbc0404e4.png)
![image](https://user-images.githubusercontent.com/44978587/213793678-e542b2c5-1bde-406e-adee-bce140310fb4.png)
![image](https://user-images.githubusercontent.com/44978587/213793708-0df09d59-3e16-4423-bd40-b2ad9fd8a1a0.png)

**Comment by Eranziel** [comment]
This is 90% fixed in the last few releases of 1.X, but I notice that mods no longer apply changes to weapons in the 2.0 branch.

**Comment by msprijatelj** [comment]
I think I accidentally made a semi-duplicate of this when I realized that the added tags aren’t detected by attack prompts in #538

**Comment by Eranziel** [comment]
They're certainly related, but I don't mind having two issues. They are two things that I'd like to see working properly. :wink:
